### ISSUE TITLE
Unsound null safety programs can we run without error

### ISSUE THREAD

**Issue by mit-mit** [dart-lang/sdk#43672]
When a null safety entry point depends on a non-migrated package, it should emit an error when run. This doesn't seem to happen any more (but I *think* it did earlier). Repro steps:

1. Git clone https://github.com/mit-mit/null-safety-test
1. `cd test_app`
1. `dart pub get`
1. `dart --enable-experiment=non-nullable run bin/test_app_unsound.dart`

Expected result: Error that app is unsound and requires `--no-sound-null-safety` flag
Actual result: Runs without error

Tested with Dart 2.10.0 stable.

**Comment by mit-mit** [comment]
cc @johnniwinther, is this maybe a regression in the CFE?

**Comment by johnniwinther** [comment]
I'll look into it.

**Comment by johnniwinther** [comment]
I get
```
dart.exe  --enable-experiment=non-nullable run bin/test_app_unsound.dart
Error: Cannot run with sound null safety as one or more dependencies do not
support null safety:

 - package:test_pkg_unmigrated

Run 'pub outdated --mode=null-safety' to determine if versions of your
dependencies supporting null safety are available.
```

**Comment by johnniwinther** [comment]
I get the same result at `2.10.0`.

**Comment by johnniwinther** [comment]
I also get the same result running the released `2.10.0` (previous attempt was run from the repo).

**Comment by mit-mit** [comment]
So do I now; I have no idea what happened, sorry about that.
